### ISSUE TITLE
write the adjusted compaction_readahead_size to LOG

### ISSUE THREAD

**Issue by mdcallag** [facebook/rocksdb#13252]
### Expected behavior

The fix for [bug 12038](https://github.com/facebook/rocksdb/issues/12038) reduces the run-time value used for compaction_readahead_size from the configured value when max_sectors_kb is smaller. But nothing is written to LOG to indicate this has been done. To make life easier for people who debug RocksDB such changes should be written to LOG, and written near the start of LOG adjacent to all of the other config options.

### Actual behavior

Nothing is written to LOG.

### Steps to reproduce the behavior

Run this long enough that compaction will start:
```
./db_bench --benchmarks=overwrite --num=$large-value
```

**Comment by mdcallag** [comment]
The code [is here](https://github.com/facebook/rocksdb/blob/02b4197544f758bdf84d80fe9319238611848c48/env/fs_posix.cc#L935-L948), but if logging is done here then that happens during compaction and I want this logged at startup with the other config options.

**Comment by hx235** [comment]
Yeah this is a bit tricky because in the setup time we haven't changed this value. It's a copy per file options now within each compaction job. Given the possibility that we will revert the change of explicit system call as mentioned in https://github.com/facebook/rocksdb/issues/12038 and that could've obsoleted this issue, I will postpone this change.
